### PR TITLE
Merge tutorial into doc

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/ingest/data-ingestion.md [hunk 1]
```diff
@@ -7,16 +7,18 @@ slug: /data-ingestion
   <link rel="canonical" href="https://docs.risingwave.com/docs/current/data-ingestion/" />
 </head>
 
-You can ingest data into RisingWave in two ways:
+In databases, users often use the `INSERT` statement to input data. However, in stream processing, data is continuously imported from upstream systems, and evidently, the `INSERT` statement is unable to meet this need. RisingWave allows users to directly create `table` and `source` to import upstream data. When there is new data entering from the upstream systems, RisingWave will directly consume the data and carry out incremental computations.
```
Comment (hengm3467) [inline docs/ingest/data-ingestion.md]: "input data" sounds a bit unnatural. Please consider rewording.

File: docs/transform/overview.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+slug: /transform-overview
+title: Overview
+---
+<head>
+  <link rel="canonical" href="https://docs.risingwave.com/docs/current/transform-overview/" />
+</head>
+
+Data transformation is a critical process in converting raw data into valuable insights. It involves applying various operations such as filtering, aggregating, and joining data to derive meaningful information. The "Transform & query data" part will dive into the techniques used in the process of transforming and querying data. Before that, here is a simple introduction to data transformation for you.
+
+## How transformation is performed
```
Comment (ShanlanLi) [inline docs/transform/overview.md]: This part was drafted by me, not from tutorial, thus need to check.

File: sidebars.js [hunk 3]
```diff
@@ -180,6 +180,21 @@ const sidebars = {
           label: "Overview",
           id: "ingest/data-ingestion",
         },
+        {
```
Comment (ShanlanLi) [inline sidebars.js]: The sidebar has a big change. The local test page will make it easier to see this change.

File: docs/ingest/include-clause.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 id: include-clause
-title: INCLUDE clause
+title: Ingest additional fields with INCLUDE clause
```
Comment (ShanlanLi) [inline docs/ingest/include-clause.md:3]: In sidebar.js, the lablel is "Ingest additional fields with INCLUDE clause", so renamed it.

File: docs/ingest/source-types.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+---
```
Comment (ShanlanLi) [inline docs/ingest/source-types.md]: This part came from the tutorial. I think the codes in it need to be checked if they need to be updated.

File: docs/ingest/source-types.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+ username = 'root',
+ password = '123456',
+ database.name = 'mydb',
+ table.name = 'orders',
+);
+```
+
+Method 1 is suitable for users who have already established standard CDC pipelines via message queues, while Method 2 is suitable for users who haven't implemented CDC via message queues or prefer a simplified architecture. Regardless of the chosen method for loading CDC data, RisingWave ensures the correct import of full and incremental data from the source table.
+
+It is worth noting that RisingWave is actively expanding the functionality and performance of direct CDC connectors. We plan to support more databases and advanced features such as full data backfill resumption, multi-table transactions, and more in the future.
```
Comment (ShanlanLi) [inline docs/ingest/source-types.md]:
```suggestion
It is worth noting that RisingWave is actively expanding the functionality and performance of direct CDC connectors. We plan to support more databases and advanced features in the future.
```

Comment (ShanlanLi) [reply docs/ingest/source-types.md]: I think this part is outdated.

File: docs/ingest/supported-sources-and-formats.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
+---
```
Comment (ShanlanLi) [inline docs/ingest/supported-sources-and-formats.md:1]: This part came from the original `CREATE SOURCE` file in sql reference. I think our old `CREATE SOURCE` was like "CREATE INDEX", displaying many things that should be placed outside of sql reference. So I move the "Supported sources and formats" in "CREATE SOURCE" to here.

File: docs/transform/ad-hoc.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+---
+id: ad-hoc-queries
+slug: /ad-hoc-queries
+title: Ad-hoc queries
+---
+<head>
+  <link rel="canonical" href="https://docs.risingwave.com/docs/current/ad-hoc-queries/" />
+</head>
+
+An ad-hoc query is a query that is created on-the-fly to meet immediate and specific information needs. Unlike predefined queries, ad-hoc queries are generated in real-time based on the your current requirements. Ad-hoc queries are commonly used in data analysis, decision-making, and exploratory data tasks where flexibility and quick access to information are essential.
```
Comment (ShanlanLi) [inline docs/transform/ad-hoc.md]: May need more introduction with examples, a bit short now.

**Review by hengm3467 — APPROVED** [review]
Made some updates. Now LGTM. Good work!

File: docs/ingest/formats-and-encode-parameters.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+- `UPSERT`: UPSERT format, where messages consumed from the message queue will perform UPSERT in RisingWave based on the primary key. To ensure UPSERT correctness, data in UPSERT format from the message queue can only be imported into RisingWave using `CREATE TABLE`.
+- `DEBEZIUM`, `MAXWELL`, `CANAL`, `DEBEZIUM_MONGO`: Mainstream Change Data Capture (CDC) formats, where messages consumed from the message queue will be processed and imported into RisingWave according to the corresponding CDC format's specification. To ensure CDC correctness, data in CDC format from the message queue can only be imported into RisingWave using `CREATE TABLE`.
+
+The `ENCODE` parameter represents the data encoding and includes the following options:
+
+- `JSON`: Data serialized in JSON format in the message queue, compatible with all `FORMAT` options.
+- `AVRO`: Data serialized in AVRO format in the message queue, compatible with all `FORMAT` options.
+- `Protobuf`: Data serialized in Protobuf format in the message queue, compatible with `FORMAT PLAIN / UPSERT / CANAL`.
```
Comment (xiangjinwu) [inline docs/ingest/formats-and-encode-parameters.md:34]: For curiosity where did this line come from? It was never the case and we only supported `plain`.

Comment (ShanlanLi) [reply docs/ingest/formats-and-encode-parameters.md:34]: Oh, thanks for pointing out this! This comes from the [tutorial](https://tutorials.risingwave.com/docs/basics/connector).

Comment (xiangjinwu) [reply docs/ingest/formats-and-encode-parameters.md:34]: Where is it hosted? Could you help correct it there?

Comment (ShanlanLi) [reply docs/ingest/formats-and-encode-parameters.md:34]: Yeah, I can do this, will copy the change.

**Pull request merged**